repos/system_upgrade/el7toel8/models/persistentnetnamesfacts.py

```diff
@@ -0,0 +1,44 @@
+from leapp.models import Model, fields
+from leapp.topics import SystemInfoTopic
+
+class PCIAddress(Model):
+    topic = SystemInfoTopic
+
+    domain = fields.String()
+    bus = fields.String()
+    function = fields.String()
+    device = fields.String()
+
+class Interface(Model):
+    topic = SystemInfoTopic
+
+    name = fields.String()
+    devpath = fields.String()
+    driver = fields.String()
+    vendor = fields.String()
+    pci_info = fields.Model(PCIAddress)
+    mac = fields.String()
+
+    def generate_link_file(self):
```

Comment:
you cannot put any methots under Models - We are using the "class" because there is not better way to do that, but you should not think about models like about objects in usual way.